packages/zapp/console/src/components/Executions/NodeExecutionCacheStatus.tsx

```diff
@@ -68,20 +74,78 @@ export interface NodeExecutionCacheStatusProps {
  * the cache status with a descriptive message. For `Core.CacheCatalogStatus.CACHE_HIT`,
  * it will also attempt to render a link to the source `WorkflowExecution` (normal
  * variant only).
+ *
+ * For Map Tasks, we will check the NodeExecutionDetail for the cache status instead. Since map tasks
+ * cotains multiple tasks, the logic of the cache status is different.
  */
 export const NodeExecutionCacheStatus: React.FC<NodeExecutionCacheStatusProps> = ({
-  taskNodeMetadata,
+  execution,
   variant = 'normal',
+}) => {
+  const taskNodeMetadata = execution.closure?.taskNodeMetadata;
+  const detailsContext = useNodeExecutionContext();
+  const [nodeDetails, setNodeDetails] = React.useState<NodeExecutionDetails | undefined>();
+
+  React.useEffect(() => {
+    let isCurrent = true;
+    detailsContext.getNodeExecutionDetails(execution).then((res) => {
+      if (isCurrent) {
+        setNodeDetails(res);
+      }
+    });
+    return () => {
+      isCurrent = false;
+    };
+  });
+
+  console.log('testiung', nodeDetails);
+  if (nodeDetails?.taskTemplate?.type === TaskType.ARRAY) {
```

Comment:
You need to use `isMapTaskType(nodeDetails?.taskTemplate?.type)` check here, from `packages/zapp/console/src/models/Task/utils.ts`
Map tasks can have three different types:
ARRAY, ARRAY_AWS, ARRAY_K8S

Reply:
fixed!